### PR TITLE
Explicit seed setting in worker processes

### PR DESCRIPTION
**Pull request by topepo**
For models that have `seed` arguments, parallel processing will result in inconsistent results because the worker seeds are not controlled. This PR generates a set of seeds immediately when `tune_grid()` or `fit_resamples()` are called, then sets them in the `iter_*()` functions. This is the same approach as `caret`. Unit tests will be in the `extratests` package. 

For example: 

```r
library(tidymodels)
data(ames)
ames <- mutate(ames, Sale_Price = log10(Sale_Price))

set.seed(123)
ames_split <- initial_split(ames, prob = 0.80, strata = Sale_Price)
ames_train <- training(ames_split)
ames_test  <-  testing(ames_split)

## -----------------------------------------------------------------------------

rf_model <-
  rand_forest(trees = 1000) %>%
  set_engine("ranger") %>%
  set_mode("regression")

rf_wflow <-
  workflow() %>%
  add_formula(
    Sale_Price ~ Neighborhood + Gr_Liv_Area + Year_Built + Bldg_Type +
      Latitude + Longitude) %>%
  add_model(rf_model)

set.seed(55)
ames_folds <- vfold_cv(ames_train, v = 10)

keep_pred <- control_resamples(save_pred = TRUE)

## -----------------------------------------------------------------------------

set.seed(130)
rf_res_seq_1 <-
  rf_wflow %>%
  fit_resamples(resamples = ames_folds, control = keep_pred)

set.seed(130)
rf_res_seq_2 <-
  rf_wflow %>%
  fit_resamples(resamples = ames_folds, control = keep_pred)

## -----------------------------------------------------------------------------

library(doParallel)
cl <- makePSOCKcluster(10)
registerDoParallel(cl)

set.seed(130)
rf_res_par_1 <-
  rf_wflow %>%
  fit_resamples(resamples = ames_folds, control = keep_pred)

set.seed(130)
rf_res_par_2 <-
  rf_wflow %>%
  fit_resamples(resamples = ames_folds, control = keep_pred)

## -----------------------------------------------------------------------------

all.equal(bind_rows(rf_res_seq_1$.metrics),
          bind_rows(rf_res_seq_2$.metrics))
# CRAN tune results:
# [1] TRUE

# new results
# [1] TRUE

all.equal(bind_rows(rf_res_par_1$.metrics),
          bind_rows(rf_res_par_2$.metrics))
# CRAN tune results:
# [1] "Component “.estimate”: Mean relative difference: 0.001654902"

# new results
# [1] TRUE

all.equal(bind_rows(rf_res_seq_1$.metrics),
          bind_rows(rf_res_par_1$.metrics))

# CRAN tune results:
# [1] "Component “.estimate”: Mean relative difference: 0.001052047"

# new results
# [1] TRUE
```

### CODE REVIEW

File: R/resample.R [hunk 1]
```diff
@@ -147,6 +147,9 @@ resample_workflow <- function(workflow, resamples, metrics, control) {
   check_rset(resamples)
   check_workflow(workflow)
   metrics <- check_metrics(metrics, workflow)
+  has_formula <- has_preprocessor_formula(workflow)
```
Comment (DavisVaughan) [inline R/resample.R:150]: Why is this here?

Comment (topepo) [reply R/resample.R:150]: Most likely vestigial code. It's not used so I'll remove it.
